### DISCUSSION TITLE
Clean-up of the bin

### DISCUSSION THREAD

**Discussion by f14m07** [liedekef/events-made-easy#537] [Q&A] [answered]
Hi,
Just a question, is there a date beyond which an approved event or booking that has been sent to the bin is permanently deleted from the bin? It seems that a clean-up is done, but I can't see where the setting is.

**Comment by liedekef** [comment]
see the GDPR (data protection) settings, cleanup is defined in there.

**Comment by f14m07** [comment]
I don't see a setting for the recycle bin. My settings are below, so an event, or an approved booking, placed in the recycle bin should never be deleted?
<img width="347" alt="image" src="https://github.com/liedekef/events-made-easy/assets/113473805/82e90247-2995-4c62-8cd0-298d1c28fe20">

**Reply by liedekef** [reply]
Those seem empty so as far as my code goes, nothing should get removed from the bin. In fact only the "remove old events" has any impact on the bin (currently next to "cleanup of people" in the manual cleanup section, but that will change in the next version because that didn't take the bin into account as it should).

**Reply by f14m07** [reply]
Thanks for the quick answer. If an event is deleted from admin, and there are still approved bookings at the time of deletion, are the bookings sent to the bin or permanently deleted?

**Reply by liedekef** [reply, marked as answer]
if you remove an event from the bin, then of course all related bookings are removed too. However: in the regular admin interface (when not in the bin), you can not hard delete events but just move them to the bin, which also moves linked bookings to their respective bin (with an option to send a mail to the people that made those bookings).

**Comment by f14m07** [comment]
Thanks. It's ok for me
